servo-media/src/backends/gstreamer/audio_decoder.rs

```diff
@@ -97,11 +104,15 @@ impl AudioDecoder for GStreamerAudioDecoder {
                 let sink = gst::ElementFactory::make("appsink", None).ok_or(())?;
                 let appsink = sink.clone().dynamic_cast::<AppSink>().map_err(|_| ())?;
 
-                // XXX Issue #18 channels support.
-                let audio_info = gst_audio::AudioInfo::new(gst_audio::AUDIO_FORMAT_F32, 44100, 1)
-                    .build()
-                    .ok_or(())?;
-                appsink.set_caps(&audio_info.to_caps().unwrap());
+                appsink.set_caps(&gst::Caps::new_simple(
+                    "audio/x-raw",
+                    &[
+                        ("format", &gst_audio::AUDIO_FORMAT_F32.to_string()),
+                        ("layout", &"interleaved"),
+                        ("channels", &(options.channels as i32)),
+                        ("rate", &(options.sample_rate as i32)),
```

Comment:
Well if you only want to handle one specific number of channels/rate here you can and probably should continue using `AudioInfo`. The full caps API is only needed here if you want to support e.g. ranges of channels (say, 1-16 channels) or rates